File: prisioner/java/src/main/java/com/everis/competition/jail/PrisonerAlwaysConfess.java
package com.everis.competition.jail;

import java.util.ArrayList;

import com.everis.competition.prisoner.Decision;
import com.everis.competition.prisoner.Prisoner;

public class PrisonerAlwaysConfess extends Prisoner {

    public PrisonerAlwaysConfess(String name) {
        super(name);
    }

    @Override
    public int decide(ArrayList<Decision> decisions) {
    	return Decision.CONFESS;
    }
}

File: prisioner/js/jail/PrisonerAlwaysConfess.js
var Prisoner = require('../prisoner/Prisoner').Prisoner;
var DECISION_OPTIONS = require('../prisoner/Decision').DECISION_OPTIONS;
var Decision = require('../prisoner/Decision').Decision;

var PrisonerAlwaysConfess = function(name) {
    Prisoner.call(this, name); 
}
PrisonerAlwaysConfess.prototype = Object.create(Prisoner.prototype);
Object.defineProperty(PrisonerAlwaysConfess.prototype, 'constructor', {
    value: PrisonerAlwaysConfess,
    enumerable: false,
    writable: true
});
PrisonerAlwaysConfess.prototype.decide = function(decisions){
    return DECISION_OPTIONS.CONFESS;
}

module.exports.PrisonerAlwaysConfess = PrisonerAlwaysConfess;